src/other/Opportunity.js

```diff
@@ -0,0 +1,421 @@
+                });
+        }
+    });
+    Opportunity.prototype.publish("removeColumn", null, "string", "number of columns in a table", function() { return this.columns(); }, { tags: ["Basic", "Shared"],
+        editor_input: function(context, widget, cell, param) {
+            cell.append("button")
+                .attr("id", context.id() + "_removeColumn" + param.id)
+                .classed("property-input custom-editor-input removeColumn", true)
+                .text("RemoveColumn")
+                .on("click", function() {
+                    widget.groupCount = widget.groupCount - 1;
+                    var new_value_after_click = "Removed a column";
+                    context.setProperty(widget, param.id, new_value_after_click);
+                });
+        }
+    });
+    Opportunity.prototype.publish("headerLabels", [], "propertyArray", "Source Columns", null, { autoExpand: Column });
+    Opportunity.prototype.publish("mouseHover", [], "propertyArray", "mouse hover options", null, { autoExpand: mouseHoverColumn });
+    Opportunity.prototype.publish("columnData", [], "propertyArray", "column data", null, { autoExpand: columnDropdown });
+    Opportunity.prototype.getIds = function () {
+        var dropdownList = this.columns();
+        dropdownList.unshift("default");
+        return dropdownList;
+    };
+    Opportunity.prototype.enter = function (domNode, element) {
+        SVGWidget.prototype.enter.apply(this, arguments);
+        var paddingTop = 30;
+        var nodeRectHeight = 14;
+        var verticalPadding = 10;
+        var h = (this.data().length + 1) * (nodeRectHeight + verticalPadding + 1) + paddingTop;
+        this.svg = element.append("g")
+            .attr("width", ((this.groupCount * 100) + 1))
+            .attr("height", h)
+        ;
+        this.svg.append('defs').append('marker')
+            .classed("arrowhead", true)
+            .attr('id', 'end-arrow')
+            .attr('viewBox', '0 -5 10 10')
+            .attr('refX', 6)
+            .attr('markerWidth', 8)
+            .attr('markerHeight', 8)
+            .attr('orient', 'auto')
+            .append('svg:path')
+            .attr('d', 'M0,-5L10,0L0,5')
+            .attr('fill', 'rgb(100,100,100)')
+        ;
+    };
+    Opportunity.prototype.update = function (domNode, element) {
+        SVGWidget.prototype.update.apply(this, arguments);
+        var context = this;
+        var data = this.data();
+        var dropDownOption = this.opportunityId();
+        data.sort(function(a, b) {
+            if (a.cur_group > b.cur_group) return 1;
+            else if (a.cur_group < b.cur_group) return -1;
+            else return 0;
+        });
+        var groups = [];
+        for (var i = 1; i <= context.groupCount; i++) {
+            groups.push(i);
+        }
+        var div = d3.select("body").append("div")
+            .attr("class", "tooltip")
+            .style("opacity", 0);
```

Comment:
This will append a new tooltip div _every_ update cycle (should be in the "enter" method)